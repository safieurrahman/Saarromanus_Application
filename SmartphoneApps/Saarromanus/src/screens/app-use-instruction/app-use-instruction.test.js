import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import AppUseInstructionScreen from './index';

const mockStore = configureStore();

const initialState = {};

let wrapper;
let store;
let component;

beforeAll(() => {
	store = mockStore(initialState);
	wrapper = shallow(<AppUseInstructionScreen store={store} />).dive();
	component = wrapper.dive();
});

describe('<AppUseInstructionScreen />', () => {
	it('should render properly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should mount a root level <ScrollView /> component', () => {
		expect(component.find(`[data-test-id='mainContainer']`).length).toBe(1);
	});

	describe('Without Data', () => {
		it('should not render any <View /> as there is not data', () => {
			expect(component.find('View').length).toBe(0);
		});
	});
});
