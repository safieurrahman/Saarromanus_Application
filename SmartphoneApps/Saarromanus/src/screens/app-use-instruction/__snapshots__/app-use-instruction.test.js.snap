// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppUseInstructionScreen /> should render properly 1`] = `
<AppUseInstructionScreen
  getAppUseInstruction={[Function]}
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
